code/clue.js

```diff
@@ -6,54 +6,271 @@ const mrGreen = {
+  weapon: '',
+  room: ''
+}
+
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach((suspect) => {
+    const randomWeapon = randomSelector(weapons)
+    suspect.favouriteWeapon = randomWeapon.id;
+  })
+}
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  document.querySelector('.killer-deck').classList.add('shakeX');
+  
+  setTimeout(function() {
+    document.querySelector('.killer-deck').classList.add('bounce');
+  }, 1000);
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  setTimeout(function() {
+    document.querySelector('.killer-deck').classList.remove('shakeX');
+    document.querySelector('.killer-deck').classList.remove('bounce');
+    document.querySelector('.killerIcon').classList.add('hide');
+
+    shuffleFavouriteWeapon();
+    mystery.killer = randomSelector(suspects)
+    document.getElementById('killerCard').style.background = mystery.killer.color
+    document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    document.getElementById('killerAge').innerHTML = `${mystery.killer.age}`
+    document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+    
+    document.getElementById('killerImage').src = `${mystery.killer.image}`
+    document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`
+
+    document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`
+
+  }, 1500);
+}
+
+const pickWeapon = () => {
+  document.querySelector('.weapon-deck').classList.add('shakeX');
+  
+  setTimeout(function() {
+    document.querySelector('.weapon-deck').classList.add('bounce');
+  }, 1000)
+  
+  setTimeout(function() {
+    document.querySelector('.weapon-deck').classList.remove('bounce');
+    document.querySelector('.weapon-deck').classList.remove('shakeX');
+
+    const favouriteWeapon = weapons.find((item) => {
+        return item.id === mystery.killer.favouriteWeapon;
+      }
+    )
+    const weaponsClone = [
+      ...weapons,
+      favouriteWeapon,
+      favouriteWeapon,
+      favouriteWeapon,
+      favouriteWeapon,
+      favouriteWeapon,
+    ]
+    mystery.weapon = randomSelector(weaponsClone);
+  
+    document.getElementById('weaponName').innerHTML = `${mystery.weapon.name} of ${mystery.weapon.weight} pounds`
+  }, 1500);
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const pickRoom = () => {
```

Comment:
A good thing to add on your pickRoom, pickWeapon, pickKiller function is to force them to be open in the right order, otherwise your logic will be broken if I open weapon card before I open the killer card and so on. 

One solution could be to give them a false value and return the value as true once the card is clicked. 
Example is to have variables at top of the script and give them a boolean value of false. 
killerPicked = false
weaponPicked = false
roomPicked = false

once the code is invoked the value is changed to true "killerPicked = true" (which is written at the bottom of the pickKiller function.